src/main/java/InfrastructureManager/Modules/NetworkStructure/Network.java

```diff
@@ -237,50 +236,222 @@ else if(update == Network.Update.DELETE) {
+	 * @param applicationId
+	 */
+	public Location getCurrentApplicationLocation(String id) {
+		Location location = null;
+		if (id.equals(this.getApplicationInstanceDeviceRelationFromList(id).getApplication().getApplicationType().getApplicationId())) {
+			location = this.getDeviceLocationRelationFromList(this.getApplicationInstanceDeviceRelationFromList(id).getApplicationDevice().getDeviceId()).getLocation();
+		}
+		return location;
 	}
+
+	/**
+	 * Function to calculate distance between two locations.
+	 * @param sourceLocation and destinationLocation
+	 * @return distance
+	 */
+	public double calculateDistance(Location source,Location destination)
+	{
+		double earthRadius = 6371.01; //Kilometers
+		double distance = 0;
+		if ((source.getLatitude() == destination.getLatitude()) && (source.getLongitude() == destination.getLongitude())) {
+			distance =  0;
+		}
+
+		else {
+			double theta = source.getLongitude() - destination.getLongitude();
+			double sourceLatitudeRadians = Math.toRadians(source.getLatitude());
+			double destinationLatitudeRadians = Math.toRadians(destination.getLatitude());
+			distance = Math.acos(Math.sin(sourceLatitudeRadians)*Math.sin(destinationLatitudeRadians) + Math.cos(sourceLatitudeRadians)*Math.cos(destinationLatitudeRadians)*Math.cos(Math.toRadians(theta)));
+			distance = earthRadius * distance;     
+		}
+		return distance;
+	}
+
+	/**
+	 * Function to get physical distance between devices
+	 * @param sourceId and destinationId
+	 */
+	public double getphysicalDistanceBetweenDevices(String sourceId, String destinationId) {
+		Location sourceLocation,destinationLocation;
+		sourceLocation = getCurrentDeviceLocation(sourceId);
+		destinationLocation = getCurrentDeviceLocation(destinationId);
+		double distance = this.calculateDistance(sourceLocation,destinationLocation);
+		return distance;
+	}
+
+	/**
+	 * Function to get physical distance between application instances.
+	 * @param sourceId and destinationId
+	 */
+	public double getphysicalDistanceBetweenApplications(String sourceId, String destinationId) {
+		Location sourceLocation,destinationLocation;
+		sourceLocation = getCurrentApplicationLocation(sourceId);
+		destinationLocation = getCurrentApplicationLocation(destinationId);
+		double distance = this.calculateDistance(sourceLocation,destinationLocation);
+		return distance;
+	}
+
+	/**
+	 * Function to get connection index from connectionList using id.
+	 * @param id
+	 */
+	public int getIndexFromConnectionList(String id) {
+		int idx=0;
+		for (Object obj : this.connectionList) {
+			Connection connection = (Connection)obj;
+			if (connection.getConnectionId().equals(id))
+			{
+				idx = this.connectionList.indexOf(obj);
+			}
+		}
+		return idx;
+	}  
+	
+	/**
+	 * Function to get network distance between devices or between application instances.
+	 * @param sourceId and destinationId
+	 */
+	public double getNetworkDistance(String sourceId, String destinationId) {
```

Comment:
It would have helped me a lot if you had some explanation written of how this is supposed to provide the network distance. I don't understand how it works?